sbt-plugin/src/main/scala/kalix/sbt/KalixPlugin.scala

```diff
@@ -46,44 +48,108 @@ object KalixPlugin extends AutoPlugin {
+    }.toSeq
+
+  private val defaultJvmArgs: Seq[String] =
+    Seq("-Dkalix.user-function-interface=0.0.0.0", "-Dlogback.configurationFile=logback-dev-mode.xml")
+
+  override def projectSettings =
+    Def.settings(
       libraryDependencies ++= Seq(
         "io.kalix" % "kalix-sdk-protocol" % KalixProtocolVersion % "protobuf-src",
         "com.google.protobuf" % "protobuf-java" % "3.17.3" % "protobuf",
         "io.kalix" %% "kalix-scala-sdk-protobuf-testkit" % KalixSdkVersion % Test),
-      logConfig := "logback-dev-mode.xml",
-      run / javaOptions ++=
-        dockerComposeUtils.value.localServicePortMappings ++
-        Seq(
-          "-Dkalix.user-function-interface=0.0.0.0",
-          s"-Dlogback.configurationFile=${logConfig.value}",
-          s"-Dkalix.user-function-port=${dockerComposeUtils.value.userFunctionPort}") ++ additionalJvmArgs,
+
+      // -------------------------------------------------------------------------------------------
+      // run task
+      Compile / run / fork := true,
+      // pass Jvm Args to forked process and disable docker-compose
+      Compile / run / javaOptions ++= disableDockerCompose +: (defaultJvmArgs ++ passedKalixArgs),
+      // redefine run task in other to be able to print the warning below
+      Compile / run := Def.inputTaskDyn {
+        val logger = streams.value.log
+        logger.warn("Kalix Proxy won't start.")
+        logger.warn("--------------------------------------------------------------------------------------")
+        logger.warn("To test this application locally you should either run it using 'sbt runAll'")
+        logger.warn("or start the Kalix Proxy by hand using the provided docker-compose file.")
+        logger.warn("--------------------------------------------------------------------------------------")
+
+        val userArgs = {
+          val args = spaceDelimited("<args>").parsed
+          " " + args.mkString(" ")
+        }
+
+        Def.taskDyn {
+          Defaults
+            .runTask(Runtime / fullClasspath, Compile / run / mainClass, Compile / run / runner)
+            .toTask(userArgs)
+        }
+      }.evaluated,
+
+      // -------------------------------------------------------------------------------------------
+      // runAll task
+      Compile / runAll / fork := true,
+      // pass Jvm Args to forked process with default docker-compose settings
+      Compile / runAll / javaOptions ++= defaultJvmArgs ++ passedKalixArgs,
+
+      // workaround for https://github.com/sbt/sbt/issues/4038
+      Compile / runAll / forkOptions := {
+        val forkOpts = (Compile / run / forkOptions).value
+        forkOpts.withRunJVMOptions((Compile / runAll / javaOptions).value.toVector)
+      },
+      // redefine the runner for runAll
+      // this is needed in order to have the forkOptions applied
+      // this is a simplified clone of Defaults.runnerInit
+      Compile / runAll / runner := {
```

Comment:
This was tricky. We want different jvm args depending on the tasks that we call. 

I got strange errors with classpath not containing some akka classes. I also tried, with help from Dale, to redefine the runner in a more elegant way. But it didn't work as expected and I decide to go for a hardcoded redefinition.